runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -46,6 +46,21 @@ IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
   }
 }
 
+IntervalSet::~IntervalSet()
+{
+}
+
+IntervalSet& IntervalSet::operator=(const IntervalSet& other)
+{
+  if (_readonly) {
+    throw IllegalStateException("can't alter read only IntervalSet");
+  }
```

Comment:
The problem is that IntervalSet assignment is used in ATN.cpp, LL1Analyzer.cpp and Parser.cpp, so they'd all need to change in some way. The only place that _readonly is really used in the code (as far as I can see) is in ATN.cpp where clearing the readonly flag on copy seems like the correct behaviour. This is also consistent with the copy constructor -- It clears the read only flag.

I argue that the copy constructor and the assignment operator should have the same behaviour; otherwise `IntervalSet s { other_set };` and `IntervalSet s; s = other_set;` will have different results when `other_set` is read only, and `IntervalSet s = other_set;` could go either way. This seems surprising.

Given C++'s value semantics, assigning to a read-only interval set seems like a bug. Honouring the flag like the copy constructor does makes intuitive sense to me.